# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSink.cpp
# Commit: 3418a8c2baa6
# Full Hash: 3418a8c2baa6d3e2d3246032b9a9268480bbe90a
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2019-10-03 16:42:51
# Description:
#   Bug 1585843 - Don't try to reject disconnected promise in AudioSink::Errored.  r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D48002
# ==============================================================================

diff -r ec52c8aa2b8e -r 3418a8c2baa6 dom/media/mediasink/AudioSink.cpp
--- a/dom/media/mediasink/AudioSink.cpp	Wed Oct 02 09:03:07 2019 +0000
+++ b/dom/media/mediasink/AudioSink.cpp	Thu Oct 03 07:32:29 2019 +0000
@@ -304,7 +304,7 @@
 void AudioSink::Errored() {
   SINK_LOG("Errored");
   mPlaybackComplete = true;
-  mEndedPromise.Reject(NS_ERROR_FAILURE, __func__);
+  mEndedPromise.RejectIfExists(NS_ERROR_FAILURE, __func__);
 }
 
 void AudioSink::CheckIsAudible(const AudioData* aData) {
